src/shell/shell_batch.cpp

```diff
@@ -19,275 +19,211 @@
 #include "shell.h"
 
 #include <climits>
-#include <stdlib.h>
-#include <string.h>
+#include <cstdlib>
+#include <cstring>
 
 #include "logging.h"
 #include "string_utils.h"
 
 // Permitted ASCII control characters in batch files
-constexpr uint8_t BACKSPACE = 8;
+constexpr uint8_t BACKSPACE       = 8;
 constexpr uint8_t CARRIAGE_RETURN = '\r';
-constexpr uint8_t ESC = 27;
-constexpr uint8_t LINE_FEED = '\n';
-constexpr uint8_t TAB = '\t';
-constexpr uint8_t UNIT_SEPARATOR = 31;
-
-BatchFile::BatchFile(DOS_Shell *host,
-                     char const *const resolved_name,
-                     char const *const entered_name,
-                     char const *const cmd_line)
-        : file_handle(0),
-          location(0),
-          echo(host->echo),
+constexpr uint8_t ESC             = 27;
+constexpr uint8_t LINE_FEED       = '\n';
+constexpr uint8_t TAB             = '\t';
+constexpr uint8_t UNIT_SEPARATOR  = 31;
+
+[[nodiscard]] static bool FoundLabel(std::string_view line, std::string_view label);
+
+BatchFile::BatchFile(DOS_Shell* host, const char* const resolved_name,
+                     const char* const entered_name, const char* const cmd_line)
+        : echo(host->echo),
```

Comment:
Is the `DOS_Shell` pointer changed or used as a temporary? (hopefully it can be `const` too)